sky/cli.py

```diff
@@ -1790,7 +1808,29 @@ def status(all: bool, refresh: bool, ip: bool, show_spot_jobs: bool,
                 with ux_utils.print_exception_no_traceback():
                     raise ValueError('Querying IP address is not supported '
                                      'for local clusters.')
+
             head_ip = handle.external_ips()[0]
+            if ports:
```

Comment:
We should perhaps add some error handling when clusters have different port modes. E.g., if user switches their global `config.yaml` from `loadbalancer` to `ingress` or vice-versa.

For instance, I launched one cluster with `ports: loadbalancer`, then changed my config to `ports: ingress`. On running `sky status --ports jupyter`, I get this error. It would be nice to either fix this or cleanly ask the user to not mix port modes.
```
(base) ➜  ~ sky status --ports jupyter
D 11-02 16:31:37 skypilot_config.py:157] Using config path: /Users/romilb/.sky/config.yaml
D 11-02 16:31:37 skypilot_config.py:160] Config loaded:
D 11-02 16:31:37 skypilot_config.py:160] {'kubernetes': {'ports': 'loadbalancer'}}
D 11-02 16:31:37 skypilot_config.py:166] Config syntax check passed.
D 11-02 16:31:38 skypilot_config.py:93] User config: kubernetes.ports -> loadbalancer
D 11-02 16:31:38 skypilot_config.py:157] Using config path: /Users/romilb/.sky/config.yaml
D 11-02 16:31:38 skypilot_config.py:160] Config loaded:
D 11-02 16:31:38 skypilot_config.py:160] {'kubernetes': {'ports': 'loadbalancer'}}
D 11-02 16:31:38 skypilot_config.py:166] Config syntax check passed.
Traceback (most recent call last):
  File "/Users/romilb/tools/anaconda3/bin/sky", line 8, in <module>
    sys.exit(cli())
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/utils/common_utils.py", line 311, in _record
    return f(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/cli.py", line 1163, in invoke
    return super().invoke(ctx)
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/utils/common_utils.py", line 332, in _record
    return f(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/cli.py", line 1821, in status
    port_details = provision_lib.query_ports(
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/provision/__init__.py", line 41, in _wrapper
    return impl(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/provision/kubernetes/network.py", line 162, in query_ports
    return _query_ports_for_loadbalancer(
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/provision/kubernetes/network.py", line 184, in _query_ports_for_loadbalancer
    external_ip = network_utils.get_loadbalancer_ip(
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/provision/kubernetes/network_utils.py", line 169, in get_loadbalancer_ip
    service = core_api.read_namespaced_service(service_name, namespace)
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/kubernetes/client/api/core_v1_api.py", line 25141, in read_namespaced_service
    return self.read_namespaced_service_with_http_info(name, namespace, **kwargs)  # noqa: E501
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/kubernetes/client/api/core_v1_api.py", line 25228, in read_namespaced_service_with_http_info
    return self.api_client.call_api(
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 348, in call_api
    return self.__call_api(resource_path, method,
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 180, in __call_api
    response_data = self.request(
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 373, in request
    return self.rest_client.GET(url,
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/kubernetes/client/rest.py", line 244, in GET
    return self.request("GET", url,
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/kubernetes/client/rest.py", line 238, in request
    raise ApiException(http_resp=r)
kubernetes.client.exceptions.ApiException: (404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Audit-Id': 'ba277f1b-fb86-41c8-888b-32896a6f1e9c', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Kubernetes-Pf-Flowschema-Uid': 'cf43db27-7222-4741-adea-8b7f8f0e681e', 'X-Kubernetes-Pf-Prioritylevel-Uid': '17aff9c6-0fa7-4909-a811-4644194e9001', 'Date': 'Thu, 02 Nov 2023 23:31:38 GMT', 'Content-Length': '244'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"services \"jupyter-2ea4-skypilot-loadbalancer\" not found","reason":"NotFound","details":{"name":"jupyter-2ea4-skypilot-loadbalancer","kind":"services"},"code":404}
```

Reply:
I think we can persist the mode for each cluster in the provider config during creation time. This will avoid this issue and allow users to have separate modes for separate clusters. Wdyt?

Reply:
This is fixed in https://github.com/skypilot-org/skypilot/pull/2713/commits/4ae66029fba78e452aaf6ce9aec7313e91615c66